### ISSUE TITLE
Add support for UUID v5

### ISSUE THREAD

**Issue by Niek** [colinhacks/zod#415]
It seems UUID v5 is not supported. Here's a simple test case:
```typescript
import * as z from 'zod'

const testUUID = 'b79cb3ba-745e-5d9a-8903-4a02327a7e09';
const schema = z.string().uuid()
schema.parse(testUUID)
```

Error output:
```json
ZodError: [
  {
    "validation": "uuid",
    "code": "invalid_string",
    "message": "Invalid uuid",
    "path": []
  }
]
```

Any chance to add support for this?

**Comment by Niek** [comment]
https://github.com/colinhacks/zod/blob/493220c2229c6d7748479fe7d4fd00d66e7ecf35/src/types.ts#L407

Strangely enough the regex validates correctly if run manually in Node:
```node
> 'b79cb3ba-745e-5d9a-8903-4a02327a7e09'.match(/([a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}){1}/i)[1]
'b79cb3ba-745e-5d9a-8903-4a02327a7e09'
```

BTW changing the regex to this would be more valid/stricter:
```node
> 'b79cb3ba-745e-5d9a-8903-4a02327a7e09'.match(/([a-f0-9]{8}-[a-f0-9]{4}-[1-5][a-f0-9]{3}-[89ab][a-f0-9]{3}-[a-f0-9]{12}){1}/i)[1]
'b79cb3ba-745e-5d9a-8903-4a02327a7e09'
```

**Comment by colinhacks** [comment]
Merged in `zod@3.0.0-alpha.40` 👍
